src/Domain/Creators/Interval.hpp

```diff
@@ -69,19 +71,56 @@ class Interval : public DomainCreator<1> {
     static constexpr Options::String help = {
         "The time dependence of the moving mesh domain."};
   };
+  struct BoundaryConditions {
+    static constexpr Options::String help = "The boundary conditions to apply.";
+  };
+  template <typename BoundaryConditionsBase>
+  struct UpperBoundaryCondition {
+    static std::string name() noexcept { return "UpperBoundary"; }
+    static constexpr Options::String help =
+        "Options for the boundary condition applied at the upper boundary.";
+    using type = std::unique_ptr<BoundaryConditionsBase>;
+    using group = BoundaryConditions;
+  };
+  template <typename BoundaryConditionsBase>
+  struct LowerBoundaryCondition {
+    static std::string name() noexcept { return "LowerBoundary"; }
+    static constexpr Options::String help =
+        "Options for the boundary condition applied at the lower boundary.";
+    using type = std::unique_ptr<BoundaryConditionsBase>;
+    using group = BoundaryConditions;
+  };
 
-  using options =
+  using basic_options =
       tmpl::list<LowerBound, UpperBound, IsPeriodicIn, InitialRefinement,
                  InitialGridPoints, TimeDependence>;
 
+  template <typename Metavariables>
+  using options = tmpl::conditional_t<
+      domain::BoundaryConditions::has_boundary_conditions_base_v<
+          typename Metavariables::system>,
+      tmpl::push_back<
+          basic_options,
+          LowerBoundaryCondition<
+              domain::BoundaryConditions::get_boundary_conditions_base<
+                  typename Metavariables::system>>,
+          UpperBoundaryCondition<
+              domain::BoundaryConditions::get_boundary_conditions_base<
+                  typename Metavariables::system>>>,
+      basic_options>;
+
```

Comment:
How about this pattern:

```cpp
template <typename Metavariables>
using options = tmpl::append<
    tmpl::list<LowerBound, UpperBound, IsPeriodicIn, InitialRefinement,
               InitialGridPoints, TimeDependence>,
    tmpl::conditional_t<
        domain::BoundaryConditions::has_boundary_conditions_base_v<
            typename Metavariables::system>,
        tmpl::list<LowerBoundaryCondition<
                       domain::BoundaryConditions::get_boundary_conditions_base<
                           typename Metavariables::system>>,
                   UpperBoundaryCondition<
                       domain::BoundaryConditions::get_boundary_conditions_base<
                           typename Metavariables::system>>>,
        tmpl::list<>>>;
```

I find this easier to understand, and with this pattern it's easier to extend a list with more conditionals.

Reply:
I find this much harder to understand. Since this is getting removed after all the systems have boundary conditions anyway, I don't see it as being worth the effort to change in every domain creator.

If we want to support domains without systems we should probably completely rethink afterwards, since this approach is admittedly tedious but I couldn't think of a better way.